rust/rsc/src/rsc/main.rs

```diff
@@ -213,7 +217,55 @@ fn launch_eviction(conn: Arc<DatabaseConnection>, tick_interval: u64, deadline:
                 .await
                 .unwrap();
 
-            tracing::info!(%res.rows_affected, "Performed TTL eviction tick.");
+            tracing::info!(%res.rows_affected, "Performed Job TTL eviction tick.");
+        }
+    });
+}
+
+fn launch_blob_eviction(
+    conn: Arc<DatabaseConnection>,
+    tick_interval: i64,
+    deadline: i64,
+    blob_stores: HashMap<Uuid, Arc<dyn blob::DebugBlobStore + Sync + Send>>,
+) {
+    // TODO: This should probably be a transaction so that a job can't add a new reference to a
+    // blob as we are deleting it.
+
+    tokio::spawn(async move {
+        let mut interval = tokio::time::interval(std::time::Duration::from_secs(
+            tick_interval.try_into().unwrap(),
+        ));
+        loop {
+            interval.tick().await;
+
+            // Blobs must be at least this old to be considered for eviction.
+            // This gives clients time to reference a blob before it gets evicted.
+            let deadline = (Utc::now() - Duration::seconds(deadline)).naive_utc();
+
+            let blobs = blob_store_service::fetch_unreferenced_blobs(conn.as_ref(), deadline)
+                .await
+                .unwrap();
+            let blob_ids: Vec<Uuid> = blobs.iter().map(|blob| blob.id).collect();
+            let mut affected: u64 = blob_ids.len().try_into().unwrap();
+
+            if affected == 0 {
+                tracing::info!(%affected, "Performed Blob TTL eviction tick.");
+                continue;
+            }
```

Comment:
If we remove the early "return" do the remaining functions handle the empty Vec correctly? Not sure its a good tradeoff here but the simplicity of this function just being straight line code might be nice.

Reply:
not currently no. `delete_blobs_by_ids` calls SeaORMs `delete_many` which panics if you give it 0 things to delete.

I could probably add a case to that function to skip on 0 ids which would be cleaner